tonic/src/client/grpc.rs

```diff
@@ -248,7 +248,12 @@ impl<T> Grpc<T> {
             })
             .map(BoxBody::new);
 
-        let mut request = request.into_http(uri, SanitizeHeaders::Yes);
+        let mut request = request.into_http(
+            uri,
+            http::Method::POST,
+            http::Version::HTTP_2,
+            SanitizeHeaders::Yes,
+        );
```

Comment:
Im not following why these values are not just hard coded into`into_http` since they should always be POST and HTTP2 and can be overridden outside (aka for grpc-web).

Reply:
Thanks for taking the time to review! Firstly, see my response on issue #911 to make sure we are on the same page.

If Rust supported default parameters, then I would propose making POST and HTTP2 defaults for `Request::into_http`.

But one issue here is that `Request::from_http` will happily accept HTTP requests of any version and method, silently dropping that information and creating subtle bugs like this one. By adding method and version to `Request::into_http`, we are making it more explicit that method and version are not preserved.

In any case, `Request::into_http` only has two callers currently, and it is restricted to the crate, so the stakes aren't very high here. I could leave `grpc.rs` alone and instead write a new method `Request::from_http_with_custom_method` if you think that would be cleaner.

Reply:
Yup, thanks for writing that up it all makes sense!